### PR TITLE
Generate TypeDoc via workflow

### PR DESCRIPTION
**Pull request by Shinigami92**
## PR Checklist

- [x] Please add a description of your changes.
- [ ] We need your changes to come with unit tests in order to keep this project in quality and easy to maintain.
- [ ] If your changes contain any new feature please be sure to document it.
- [ ] Please add a link to the open issue or task that this pull request will resolve.
- [ ] Add a note to the changelog to indicate the change.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

If yes please indicate it and migration info into the CHANGELOG.md file.

---

Generate TypeDoc via workflow

### CODE REVIEW

**Comment by Shinigami92** [comment]
I currently set it up so that someone of us need to manually perform the GH action

File: .github/workflows/generate-typedoc.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+      - name: Generate TypeDoc
+        run: |
+          npm ci
+          npm run typedoc
+
+      - name: Commit files
+        if: "github.repository == 'pioardi/poolifier' && github.ref == 'refs/heads/master'"
+        run: |
+          git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
```
Comment (jerome-benoit) [inline .github/workflows/generate-typedoc.yml:30]: Does the number have a special meaning? Or it's just random?

**Comment by jerome-benoit** [comment]
Looks good.

**Review by jerome-benoit — APPROVED** [review]

**Comment by pioardi** [comment]
I guess we should mention to avoid to publish any doc manually in our contribution guidelines

File: .github/workflows/generate-typedoc.yml (hunk 1 above)
Comment (Shinigami92) [reply .github/workflows/generate-typedoc.yml:30]: It's coming from the examples: https://github.com/ad-m/github-push-action#example-workflow-file

**Comment by Shinigami92** [comment]
> I guess we should mention to avoid to publish any doc manually in our contribution guidelines

Yeah, I would hand this over to you :slightly_smiling_face:  
You should directly write something into master as you wish

**Comment by Shinigami92** [comment]
Sadly it doesn't work because of the restriction: https://github.com/pioardi/poolifier/runs/2006304445?check_suite_focus=true#step:6:10

Need to read about this further later on

**Pull request merged**